File: OLP_ICPC/Python/sammyuri_redstone.py
tt = int(input())
for t in range(tt):
    n = int(input())
    gears = list(map(int, input().split()))
    valid = False
    for i in range(n):
        for j in range(i + 1, n):
            if gears[i] == gears[j]:
                valid = True
    if valid:
        print("YES")
    else:
        print("NO")

File: OLP_ICPC/C++/sammyuri_redstone.cpp
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;
#define all(x) x.begin(), x.end()
#define vecin(name, len) vector<int> name(len); for (auto &_ : name) cin >> _;
#define vecout(v) for (auto _ : v) cout << _ << " "; cout << endl;

void solve() {
	int n;
	cin >> n;
	vecin(gears, n);
	set<int> seen;
	for (auto gear: gears) {
		if (seen.count(gear)) {
			cout << "YES\n";
			return;
		}
		seen.insert(gear);
	}
	cout << "NO\n";
}

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);
	int t;
	cin >> t;
	while (t--) solve();
}